### ISSUE TITLE
Deletion call error

### ISSUE THREAD

**Issue by qingzhou1** [oicr-gsi/debarcer#94]
I just realized this newer version cause incorrect deletion calling.  Please see an example below:
                Old version                ||Newer version
Pos | Ref | DP | A | T | G | C | D || DP | A | T | G | C | D
7577556 | C | 2560 | 0 | 0 | 0 | 2560 | 0 || 2875 | 0 | 0 | 0 | 2803 | 72
7577557 | A | 2560 | 2499 | 2 | 0 | 0 | 59 || 2860 | 2858 | 0 | 0 | 2 | 0
7577558 | G | 2560 | 0 | 0 | 2560 | 0 | 0 || 2875 | 72 | 2 | 2801 | 0 | 0
7577559 | G | 2560 | 0 | 0 | 2560 | 0 | 0 || 2875 | 0 | 0 | 2875 | 0 | 0

**Comment by morgantaschuk** [comment]
Thanks for reporting it! We'll take a look.

**Comment by lheisler** [comment]
Hi qingzhou1, can you please indicate which versions you are comparing.

**Comment by qingzhou1** [comment]
My example is compare Debarcer (V0.3.1) with this current version, but I also used your previous version of this current version (I am not sure how you call it). Anyway, Debarcer (V0.3.1) and the previous version of this python version of Debarcer give me correct output (the later one, all positions were off by one base). Seems like the problem only happened in this new version. I try to fix it, one problem I found is, in the get_consensus_seq.py:
 # Next position is a deletion (current base + next bases are ref)
                            elif read.indel < 0:
                                ref_base = ref_seq[ref_pos:ref_pos + abs(read.indel) + 1]
                                alt_base = read_data.query_sequence[read.query_position]
If you doing this, and in the generate_consensus.py: 
for allele in cons:
                            # ref > 1 => deletion
                            if len(allele[0]) > 1:
                                counts['D'] += cons[allele]
I am sure you count deletion 1 base early. But I cannot fix why there is always a mismatch 1 base later.
I am not sure if my thoughts are right.

**Comment by lheisler** [comment]
Thanks qingzhou,

We had caught on off-by-one error that was corrected in the most recent version.  I'll review to ensure that this had been addressed properly.

**Comment by rjovelin** [comment]
The one-off error has been corrected.

I also compared indel calls between debarcer and samtools mpileup. The results are generally consistent, although not always. For some positions deletion calls with debarcer can be greater or lower than with samtools, with the limitations of my test data that the targeted regions are small and that indel calls overall are low. 

Results from 1 sample with targeted sequencing across 5 chromosomes. 

- 98% (766/782) of the positions have the same deletion counts between debarcer and samtools
- 99.5% (778/782) of the positions have the same insertion counts between debarcer and samtools
- deletion count differences range from 1 to 7 reads
- maximum insertion count difference is 1 read

I suspect that differences result from filtering settings that differ between pysam and samtools, making it difficult to compare the 2 approaches directly. I am closing this issue because  results are generally very similar.
